app/Filament/Resources/ApiTokenResource.php

```diff
@@ -0,0 +1,158 @@
+                    ->descriptions([
+                        'results:read' => 'Allow this token to read results.',
+                        'speedtests:run' => 'Allow this token to run speedtests.',
+                        'ookla:list-servers' => 'Allow this token to list servers.',
+                    ]),
+
+                DateTimePicker::make('expires_at')
+                    ->label('Expires at')
+                    ->nullable()
+                    ->native(false)
+                    ->helperText('Leave empty for no expiration'),
+            ]),
+        ];
+    }
+
+    public static function form(Form $form): Form
+    {
+        return $form->schema(static::getTokenFormSchema());
+    }
+
+    public static function table(Table $table): Table
+    {
+        return $table
+            ->query(PersonalAccessToken::query()->where('tokenable_id', Auth::id()))
+            ->columns([
+                TextColumn::make('name')->searchable(),
+                TextColumn::make('abilities')->badge(),
+                IconColumn::make('is_valid')
+                    ->label('Valid')
+                    ->boolean()
+                    ->toggleable()
+                    ->state(fn ($record) => $record->expires_at === null || $record->expires_at->isFuture()),
+                TextColumn::make('created_at')
+                    ->dateTime(config('app.datetime_format'))
+                    ->timezone(config('app.display_timezone'))
+                    ->toggleable()
+                    ->sortable()
+                    ->alignEnd(),
+                TextColumn::make('last_used_at')
+                    ->dateTime(config('app.datetime_format'))
+                    ->timezone(config('app.display_timezone'))
+                    ->toggleable()
+                    ->toggledHiddenByDefault()
+                    ->sortable()
+                    ->alignEnd(),
+                TextColumn::make('expires_at')
+                    ->dateTime(config('app.datetime_format'))
+                    ->timezone(config('app.display_timezone'))
+                    ->toggleable()
+                    ->sortable()
+                    ->alignEnd(),
+            ])
+            ->filters([
+                TernaryFilter::make('valid')
+                    ->label('Token Validity')
+                    ->trueLabel('Only valid tokens')
+                    ->falseLabel('Only expired tokens')
+                    ->native(false)
+                    ->queries(
+                        true: fn (Builder $query) => $query->where(function ($q) {
+                            $q->whereNull('expires_at')
+                                ->orWhere('expires_at', '>', now());
+                        }),
+                        false: fn (Builder $query) => $query->whereNotNull('expires_at')->where('expires_at', '<=', now()),
+                        blank: fn (Builder $query) => $query,
+                    ),
+
+                MultiSelectFilter::make('abilities')
```

Comment:
MultiSelectFilter was deprecated, let's use SelectFilter instead with the `->multiple()` attribute: https://filamentphp.com/docs/3.x/tables/filters/select